substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/SerializationSupport.java

```diff
@@ -126,18 +126,26 @@ public Object addConstructorAccessor(Class<?> declaringClass, Class<?> targetCon
     }
 
     @Override
-    public Object getSerializationConstructorAccessor(Class<?> declaringClass, Class<?> rawTargetConstructorClass) {
-        Class<?> targetConstructorClass = Modifier.isAbstract(declaringClass.getModifiers()) ? stubConstructor.getDeclaringClass() : rawTargetConstructorClass;
+    public Object getSerializationConstructorAccessor(Class<?> rawDeclaringClass, Class<?> rawTargetConstructorClass) {
+        Class<?> declaringClass = rawDeclaringClass;
+
+        if (declaringClass.getName().contains("$$Lambda$")) {
+            try {
+                // Checkstyle: stop
+                declaringClass = Class.forName("java.lang.invoke.SerializedLambda");
```

Comment:
Can we fetch this once in a final field?

Reply:
Added final field `serializedLambdaClass` and used in proper places.